projects/igniteui-angular/src/lib/date-picker/date-picker.component.ts

```diff
@@ -308,6 +313,10 @@ export class IgxDatePickerComponent implements IDatePicker, ControlValueAccessor
     }
     public set disabledDates(value: DateRangeDescriptor[]) {
         this._disabledDates = value;
+        if (this._ngControl && this._ngControl.control && this._ngControl.control.validator && this._value && this._disabledDates) {
```

Comment:
This is not how we should be doing validity checks. We do not need `onChangeCallback`  here as it needs to be called only if the value of the control has changed. Instead, we should implement the `Validator` interface from `@angular/forms` and then register a validator change callback, something like this:

```typescript
private _onValidatorChange = () => { };
public registerOnValidatorChange(fn: any) {
    this._onValidatorChange = fn;
}
```
Then call that `this._onValidatorChange` right after `this_disabledDates` is updated. Everything else can be removed.

Reply:
done